plugins/org.jkiss.dbeaver.ext.mssql/src/org/jkiss/dbeaver/ext/mssql/SQLServerUtils.java

```diff
@@ -135,7 +150,45 @@ public static String getSystemSchemaFQN(JDBCDataSource dataSource, String catalo
     }
 
     public static String getSystemTableName(SQLServerDatabase database, String tableName) {
-        return SQLServerUtils.getSystemSchemaFQN(database.getDataSource(), database.getName(), SQLServerConstants.SQL_SERVER_SYSTEM_SCHEMA) + "." + tableName;
+        final SQLServerDataSource dataSource = database.getDataSource();
+        if (isDriverBabelfish(dataSource.getContainer().getDriver())) {
+            switch (tableName) {
+                case "default_constraints":
+                    return "(SELECT CAST(CONCAT('DF_',  o.relname, '_', d.oid) AS VARCHAR(20)) AS name, d.oid AS object_id, CAST(NULL AS INT) AS principal_id, o.relnamespace AS schema_id, d.adrelid AS parent_object_id, CAST('D' AS CHAR(2)) AS type, CAST('DEFAULT_CONSTRAINT' AS VARCHAR(60)) AS type_desc, CAST(NULL AS sys.datetime) AS create_date, CAST(NULL AS sys.datetime) AS modified_date, CAST(0 AS sys.bit) AS is_ms_shipped, CAST(0 AS sys.bit) AS is_published, d.adnum AS parent_column_id, pg_get_expr( d.adbin, d.adrelid ) AS definition, d.adbin, CAST(1 AS sys.bit) AS is_system_named FROM pg_catalog.pg_attrdef AS d INNER JOIN pg_catalog.pg_class AS o ON d.adrelid = o.oid) AS";
```

Comment:
Nope. This is not a good decision. getSystemTableName must return a fully qualified system table name (usually short), not a new SQL query for Babelfish only.

Reply:
@LonwoLonwo With the current Babelfish server release the queries are required (it is the only decision possible).  As the Babelfish server changes in the future the queries could be removed.

Reply:
@LonwoLonwo Would it be best to use a Babelfish-specific file for the switch statement?  The intent was to have the queries be prominent to motivate future removal of individual queries, when that is possible.

Reply:
Hi @LonwoLonwo ... I'm a product manager from Amazon. Is there someone from the Dbeaver.io company that I can speak with by email or LinkedIn? Please let me know.

Reply:
@slightst if your questions are related to DBeaver Community you can ask them in discussions on GitHub. If you need some information about the DBeaver commercial products you can send an email to support@dbeaver.com.

Reply:
Thank you, I sent an email to support@dbeaver.com. Please have a look.

Reply:
@LonwoLonwo Sorry for interrupting :) When you're free, I think @okeuday was asking you: How do you wanted the code organized?

Reply:
@LonwoLonwo I updated the commit with small additions that improve the functionality.  However, I didn't change how the queries are provided in `SQLServerUtils.java` yet.  Please tell me if the structure should be different.